### PR TITLE
Fix LZO decompression

### PR DESCRIPTION
**Pull request by Schamper**
Fixes https://github.com/fox-it/dissect.squashfs/issues/11

### CODE REVIEW

File: dissect/util/compression/lzo.py [hunk 1]
```diff
@@ -58,54 +46,49 @@ def decompress(src: Union[bytes, BinaryIO], header: bool = True, buflen: int = -
     val = src.read(1)[0]
```
[the remark this answers is not in the document]
Comment (Schamper) [reply dissect/util/compression/lzo.py:48]: This turns out to be not quite correct. See also https://docs.kernel.org/staging/lzo.html and https://github.com/torvalds/linux/blob/master/lib/lzo/lzo1x_decompress_safe.c.

[the remark this answers is not in the document]
Comment (pyrco) [reply dissect/util/compression/lzo.py:48]: Maybe good to add these references.
I looked a bit at making the code more readable, but that is far from trivial. Maybe comment the if statements a bit about which byte encoding is indicated.

**Review by pyrco — COMMENTED** [review]
see comment

File: dissect/util/compression/lzo.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Schamper) [reply dissect/util/compression/lzo.py:51]: Added them as references

File: dissect/util/compression/lzo.py [hunk 2]
```diff
@@ -56,56 +46,57 @@ def decompress(src: Union[bytes, BinaryIO], header: bool = True, buflen: int = -
...
+    if src.seek(5) == 5 and val == 17:
+        src.seek(offset)
+        _ = src.read(1)  # This would be the bitstream version but we don't currently use it
         val = src.read(1)[0]
+    else:
+        src.seek(offset)
 
-    trailing = 0
+    if val > 17:
+        dst += src.read(val - 17)
```
[the remark this answers is not in the document]
Comment (Schamper) [reply dissect/util/compression/lzo.py:61]: `state` looks unused there: https://github.com/FFmpeg/FFmpeg/blob/master/libavutil/lzo.c#L157

**Comment by Schamper** [comment]
> Would it be an idea to add some comments about which instruction encoding is used?
> Including the first byte encodings

You seem to understand the algorithm better than me by now so feel free to make some suggestions 😄

**Pull request merged**